### PR TITLE
Added the game logic and responsive resizing to the game as per specifications 

### PR DESCRIPTION
**Pull request by tarique1988**
## Summary
Followed the specifications provided in the `ReadMe.md` file and added the required features to the game.

## GamePlay
Here are some points to note about the features that were added in this update:
- The canvas takes up the entire height and width of the browser.
- This game currently has two players and a ball
- The game can be started, paused, and resumed using the `spacebar` 
- The player on the left is named "**PlayerLeft**", the one on the right is named "**PlayerRight**"
- To move **PlayerLeft**, you can use: `w` for `up` and `s` for `down`
- To move **PlayerRight**, you can use: `UpArrow` for `up` and `DownArrow` for `down`
- The x positions of both players is fixed.
- Both players must keep the ball within the gaming area. Whenever a player fails to do so, the other player will gain 10 points which will reflect in the score located above the player.
- Each time a player successfully deflects the ball, the speed of the game is increased by 5% to make the game more challenging.
- The said speed resets when a player misses.
- The ball restarts at the center after each miss.
- The ball automatically bounces off the vertical bounds.
- The game objects are responsive, i.e, the players and ball scale and adjust their position whenever the window is resized.
- As of now, the game can run indefinitely.

## Note
The code has not been tested because it is a demonstration of my ability to write the code. However, in real projects, I will be spending a considerable amount of time testing.

I look forward to your review and feedback.

Cheers!

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,46 +1,305 @@
...
+        this.playerLeft.dy = -this.gameSpeed
+      }
+      if (e.keyCode == 83 || e.which == 83) {
+        this.playerLeft.dy = this.gameSpeed
+      }
+
+      if (e.keyCode == 32 || e.which == 32) {
+        this.isStarted = !this.isStarted
+      }
+    }
```
Comment (demiculus) [inline public/index.html:252]: why do we have the same code twice?

File: public/index.html [hunk 2]
```diff
@@ -1,46 +1,305 @@
 <!DOCTYPE html>
 <html>
+
 <head>
-	<title>Ping Pong Game</title>
+  <title>Ping Pong Game | TAM</title>
```
Comment (demiculus) [inline public/index.html:5]: Whats TAM?

File: public/index.html (hunk 1 above)
Comment (tarique1988) [reply public/index.html:252]: That's a mistake on my part. Towards the end, I refactored the key handler and resize functions and moved the logic to app.keyDownHandler and so on, that standalone function is not being used anywhere, I just forgot to delete it after refactoring. My bad!

**Comment by demiculus** [comment]
If you were to improve this code, how would you do it? (making it cleaner, and easier to edit)

No need to do it, the ideas are enough.

**Review by tarique1988 — COMMENTED** [review]
> If you were to improve this code, how would you do it? (making it cleaner, and easier to edit)
> 
> No need to do it, the ideas are enough.

I would start by moving the code out of the html script tag, I kept it this way because the initial code was there and the instructions suggested that I just remove the existing code and carry on.

I would also find functions that can be refactored to be a bit more generic. It would be helpful in a bigger game. For this task, it did not make sense to make the functions generic, because there was no need to reuse them anywhere else. So everything went inside the app!

I would also divide any large functions into smaller ones, but as is the case, I already did it for many functions while working on this task.

And maybe name the functions better and write more helpful comments wherever necessary. 

And last but not the least, port this project to TS rather than JS. Types make everything more readable :)

To be very honest, this list could go on and on.

Cheers

**Pull request merged**